src/Compilers/CSharp/Portable/Binder/Semantics/Conversions/ConversionsBase.cs

```diff
@@ -1335,15 +1336,43 @@ public static bool HasIdentityConversion(TypeSymbol type1, TypeSymbol type2)
             Debug.Assert((object)type1 != null);
             Debug.Assert((object)type2 != null);
 
-            return type1.Equals(type2, TypeCompareKind.AllIgnoreOptions);
+            var compareKind = includeNullability ?
+                TypeCompareKind.AllIgnoreOptions | TypeCompareKind.CompareNullableModifiersForReferenceTypes | TypeCompareKind.UnknownNullableModifierMatchesAny :
```

Comment:
>TypeCompareKind.UnknownNullableModifierMatchesAny [](start = 111, length = 49)

If we have two types one of them has unknown nullability set and the other one doesn't. Which one do we pick during inference?

Reply:
Good catch. Added `// PROTOTYPE` comment and skipped test for now.

---
In reply to: [169812431](https://github.com/dotnet/roslyn/pull/24935#discussion_r169812431) [](ancestors = 169812431)